### PR TITLE
Website redesign

### PR DESCRIPTION
**Pull request by lewisakura**
- [X] Move typography to plain css
- [X] Fix dodgy FAQ codeblock
- [x] Navbar colors only on active one
- [x] Cloud page
- [ ] Make downloads page closer to design
- [x] Mobile support
- [x] Switch to final logo
- [x] Be cute

### CODE REVIEW

File: src/components/typography/index.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// @ts-nocheck - bugged with astro
```
Comment (Vendicated) [inline src/components/typography/index.ts:1]: no clue what this is about, what you can just use an astro file for this and put this code in the frontmatter

**Review by Vendicated — CHANGES_REQUESTED** [review]

**Comment by Vendicated** [comment]
Like I said before, we should probably make the typography stuff global styles instead, or classnames with global styles. This is especially important for Svelte stuff (/plugins should likely also become svelte) because they can't access astro components

**Comment by Vendicated** [comment]
What do you still want to change before marking this as ready? I think it would be useful to have a checklist, to track progress and maybe I can also help with some things

**Comment by lewisakura** [comment]
> What do you still want to change before marking this as ready? I think it would be useful to have a checklist, to track progress and maybe I can also help with some things

Unsure on what else really needs to be done - I'll go through the design again and make a checklist, though odds are it's probably just the docs (which we're putting off in this PR) & the cloud index page which we need to talk about.

**Comment by lewisakura** [comment]
Please review everything besides what's on the checklist already. I will append all review points to the checklist as blockers before finalising.

File: src/components/Footer.astro [hunk 2]
```diff
@@ -12,11 +12,13 @@ import { Icon } from "astro-icon";
...
     <div class="content">
         <div class="source-and-disclaimer">
-            <Link href={SOURCE_CODE}>source code</Link>
-            <span class="divider">&nbsp;•&nbsp;</span>
-            <span>
-                Discord is trademark of Discord Inc. Vencord is not affiliated
-                or endorsed by Discord Inc.
+            <span class="p-overline-s">
+                <Link href={SOURCE_CODE}>source code</Link>
+                <span class="divider">&nbsp;-</span>
```
Comment (lewisakura) [inline src/components/Footer.astro:17]: nit: remove nbsp

File: src/pages/faq.astro
Comment (lewisakura) [inline src/pages/faq.astro:1]: add support link to bottom

File: src/components/Footer.astro
Comment (lewisakura) [inline src/components/Footer.astro:1]: align svgs

File: src/components/NavBar.astro [hunk 3]
```diff
@@ -38,8 +38,11 @@ const { title } = Astro.props;
 
 <style>
     nav {
-        background-color: var(--surface);
-        padding: 1em 0;
+        background-color: var(--bg0);
+
+        position: sticky;
```
Comment (lewisakura) [inline src/components/NavBar.astro:43]: unsure if sticky nav is good, need more input

File: src/pages/404.astro
Comment (lewisakura) [inline src/pages/404.astro:1]: replace with designed 404

File: src/pages/cloud/index.astro
Comment (lewisakura) [inline src/pages/cloud/index.astro:1]: should we remove this?

File: src/pages/plugins.astro [hunk 4]
```diff
@@ -39,12 +40,15 @@ Astro.response.headers.set("Cache-Control", "s-maxage=300");
                     <div class="plugin">
                         {/*
                         <img src={plugin.screenshot || "/assets/screenshot-placeholder.png"} class="plugin-screenshot" />
-                    */}
+                        */}
```
Comment (lewisakura) [inline src/pages/plugins.astro:43]: decide what we want to do for this cause we need to get the devs to take screenshots if we go forward

File: src/components/NavBar.astro [hunk 5]
```diff
@@ -1,34 +1,34 @@
...
                 </li>
                 <li>
-                    <a href="/docs">Docs</a>
+                    <a href="/docs" style="--accent:var(--accentGreen)" class:list={{ active: page === "docs" }}>Docs</a>
                 </li>
                 <li>
-                    <a href="/cloud">Cloud</a>
+                    <a href="/cloud" style="--accent:var(--accentYellow)" class:list={{ active: page === "cloud" }}>Cloud</a>
                 </li>
             </ul>
```
Comment (lewisakura) [inline src/components/NavBar.astro:33]: improve this because this is terrible

File: src/pages/plugins.astro [hunk 6]
```diff
@@ -39,12 +40,15 @@ Astro.response.headers.set("Cache-Control", "s-maxage=300");
...
                         <div class="plugin-content">
-                            <span class="title">{plugin.name}</span>
-                            <span class="author">
-                                by {plugin.authors.map(a => a.name).join(", ")}
-                            </span>
+                            <span class="p-label-l">{plugin.name}</span>
+                            <div class="author">
+                                <span id="by">by</span>
```
Comment (Vendicated) [inline src/pages/plugins.astro:47]:
```suggestion
                                <span class="by">by</span>
```

File: src/pages/plugins.astro [hunk 7]
```diff
@@ -97,20 +106,31 @@ Astro.response.headers.set("Cache-Control", "s-maxage=300");
...
+
+        color: var(--grey1);
+    }
+
+    .plugin-content .author #by {
+        color: var(--grey0);
+        margin-right: 0.25em;
+    }
+
     .plugin-content .description {
```
Comment (Vendicated) [inline src/pages/plugins.astro:134]: what's up with these overspecific selectors? Just remove .plugin-content from everywhere

File: src/pages/faq.astro [hunk 8]
```diff
@@ -42,25 +45,33 @@ const faq = await getSortedFaq();
...
+        color: var(--grey2);
+        border-radius: 4px;
+        padding: 0.75em;
+    }
+
+    .faq-content :global(p) {
+        margin-top: 0;
+    }
+
+    .faq-content :global(*:last-child) {
```
Comment (Vendicated) [inline src/pages/faq.astro:73]: what is this for? From my experience selectors like this have bad performance

File: src/pages/faq.astro [hunk 9]
```diff
@@ -21,7 +22,9 @@ const faq = await getSortedFaq();
                     <details>
                         <summary>{f.data.title}</summary>
 
-                        <Content />
+                        <div class="faq-content">
```
Comment (Vendicated) [inline src/pages/faq.astro:25]: why does this need a wrapper div? Just use `details :global(p)` instead

File: src/pages/404.astro [hunk 10]
```diff
@@ -2,8 +2,8 @@
 import Layout from "../layouts/Layout.astro";
 ---
 
-<Layout title="404 - Page Not Found" description="Uh oh" opengraph={false}>
-    <h1>404 - Page Not Found</h1>
```
Comment (Vendicated) [inline src/pages/404.astro:6]: Why did you remove this? The average user likely doesn't know what a 404 is

File: src/layouts/Layout.astro [hunk 11]
```diff
@@ -28,17 +32,20 @@ const { title, description, opengraph } = Astro.props;
...
-                    <meta name="og:title" content={title} />
+                    <meta name="og:title" content={title ?? "Vencord"} />
                     <meta name="og:description" content={description} />
                     <meta name="og:url" content={Astro.url} />
+                    <meta name="og:image" content={new URL("/assets/favicon.png", Astro.site)} />
                 </>
             )
         }
+        {/* FIXME: Seems to be a bug in Astro because this statement doesn't work in the above block. */}
+        {(title && opengraph !== false) && <meta name="og:site_name" content="Vencord" />}
```
Comment (Vendicated) [inline src/layouts/Layout.astro:43]: I've experienced that as well lol

File: src/components/NavBar.astro [hunk 12]
```diff
@@ -62,12 +64,19 @@ const { title } = Astro.props;
...
         padding: 0;
     }
 
+    div#btnsleft {
+        flex: 1;
+    }
+
+    #logo img {
+        vertical-align: middle;
+    }
```
Comment (Vendicated) [inline src/components/NavBar.astro:78]: these selectors are overly specific again and should probably use a class instead of an id

**Review by Vendicated — CHANGES_REQUESTED** [review]

File: src/pages/faq.astro (hunk 8 above)
Comment (lewisakura) [reply src/pages/faq.astro:73]: removing the ugly margins on the last element of the faq content, otherwise it leaves a massive gap for no reason

File: src/pages/faq.astro (hunk 9 above)
Comment (lewisakura) [reply src/pages/faq.astro:25]: was trying to avoid adding more global styles than necessary

Comment (Vendicated) [reply src/pages/faq.astro:25]: details :global(p) is already scoped because the `details` selector is not global

**Review by Vendicated — APPROVED** [review]
